### ISSUE TITLE
Logits Score Always Outputs -inf During Inference on Hugging Face and vLLM

### ISSUE THREAD

**Issue by cjfcsjt** [QwenLM/Qwen2-VL#154]
I've encountered a consistent issue where the logits score returns -inf during offline inference using models from Hugging Face/vLLM, even using the default inference example. How can I solve this issue? Below is the code.

Environment: torch 2.4.0, vllm 0.5.5+cu122, flash-attn 2.6.3, transformers 4.45.0.dev0

```
MODEL_PATH = "Qwen/Qwen2-VL-7B-Instruct"
model = Qwen2VLForConditionalGeneration.from_pretrained(
    MODEL_PATH, torch_dtype="auto", device_map="auto" #, attn_implementation="flash_attention_2",
)
processor = AutoProcessor.from_pretrained(MODEL_PATH)
messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "image",
                "image": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen-VL/assets/demo.jpeg",
            },
            {"type": "text", "text": "Describe this image."},
        ],
    }
]

# Preparation for inference
text = processor.apply_chat_template(
    messages, tokenize=False, add_generation_prompt=True
)
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=[text],
    images=image_inputs,
    videos=video_inputs,
    padding=True,
    return_tensors="pt",
)
inputs = inputs.to("cuda")

# Inference: Generation of the output
generated_ids = model.generate(**inputs, max_new_tokens=128, output_scores=True, return_dict_in_generate=True)
score = generated_ids.scores
```

**Comment by nulla-dies-sine-linea** [comment]
I have the same issue. @cjfcsjt, did you manage to solve it?

**Comment by fiona-lxd** [comment]
+1

**Comment by Dineshkumar-Anandan-ZS0367** [comment]
Is there any possibility to get score of logits on vLLM for qwen2-vl-7b-instruct

**Comment by ppalantir** [comment]
> set do_sample=False

temperature > 0 only works when do_sample=True

to return logits, we cannot use temperature?
